### PR TITLE
Awesome Books

### PR DESCRIPTION
**Pull request by tta2yta**
In this project, we set up basic functionality by using Javascript.
 Based on requirements, we set up the following features:

 1. Create an object collection that keeps a list of books
 2. Create a function to add a new book to the collection
 3. Create a function to remove a book from the collection 
 4. Display all books saved in the collection
 5. data is preserved in the browser's memory by using localStorage.

Kindly review my code before I merge it.
Thanks in advance.

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  const titleInput = title.value;
+  const authorInput = author.value;
+
+  const book = { title: titleInput, author: authorInput };
+  const bookArrSize = bookArr.push(book);
+
+  createElement(titleInput, authorInput, bookArrSize);
+  localStorage.setItem('Books', JSON.stringify(bookArr));
+  window.location.reload();
+});
```
Comment (abruzy) [inline js/script.js:43]: Good job! I think you missed out on implementing a function that handles adding a new book to the array.

For example, you can have a function like this

```function addBook (title, author) {}```

and in your event listener, you can just pass the function addBook there like this

```document.addEventListener('click', addBook)```

Do this for other functions as well like ```removeBook(id)``` and ```displayBook()```

Screenshot from 2021-07-06 10-04-39

File: js/script.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const parentId = btnRemoveId.charAt(btnRemoveId.length - 1);
+  const parent = document.getElementById(parentId);
+
+  parent.remove();
+  const index = Number(parentId);
+
+  const books = bookArr.filter((book, i) => i !== index);
+  localStorage.setItem('Books', JSON.stringify(books));
+  window.location.reload();
+});
```
Comment (abruzy) [inline js/script.js:63]: kindly create a function to remove as stated above

**Review by abruzy — CHANGES_REQUESTED** [review]
```markdown
Hi @tta2yta and @enkog, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [OPTIONAL] the use of `DOMContentLoaded` over `load` "A different event, load, should be used only to detect a fully-loaded page. It is a common mistake to use load where DOMContentLoaded would be more appropriate. [see](https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: js/script.js (hunk 2 above)
Comment (tta2yta) [reply js/script.js]: Hello. Thank you for your code review.

We have updated the points you mentioned

Kindly review my code before I merge it.
Thanks in advance.

**Review by de-scholar — APPROVED** [review]
Hi @enkog and @tta2yta  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- [ ] I would suggest you to favor the use array methods that doesn't mutate the array, e.g. concat, filter, slice instead of push, pop, splice and so on. (see code example)


- [ ] I would suggest you to use unique id property for each book since it would make it easier to remove books with duplicate values.
- [ ] It would be nice if when I click on the `add` button while the form is empty, I should get an error note 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**